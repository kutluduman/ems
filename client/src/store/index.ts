import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;