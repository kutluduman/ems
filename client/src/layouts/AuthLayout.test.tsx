import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import ThemeConfig from "../theme";
import { store } from "../store";
import AuthLayout from "./AuthLayout";

describe("AuthLayout Component", () => {
  test("renders the AuthLayout Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <ThemeConfig>
            <AuthLayout>
              <h1>Some content</h1>
            </AuthLayout>
          </ThemeConfig>
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});