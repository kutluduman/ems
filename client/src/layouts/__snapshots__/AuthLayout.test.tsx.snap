// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout Component renders the AuthLayout Component 1`] = `
<header
  className="css-103o8ot"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      className="MuiBox-root css-1ngw63v"
      src="https://minimal-kit-react.vercel.app/static/logo.svg"
    />
  </a>
  <p
    className="MuiTypography-root MuiTypography-body2 css-128dl2h-MuiTypography-root"
  >
    <h1>
      Some content
    </h1>
  </p>
</header>
`;
